js/hookLiveEdit.js

```diff
@@ -0,0 +1,338 @@
+var has_been_moved = false;
+var modules_list = new Array();
+var hooks_list = new Array();
+var hookable_list = new Array();
+var timer;
+var baseDir = prestashop.urls.base_url;
+$(document).ready(function () {
+    // do some place for submit button
+    $('body').css('margin-top', '45px');
+    // this is the result box, which will "pop up" the succeed or fail result
+    $('#fancy').fancybox({
+        autoDimensions: true,
+        autoSize: false,
+        autoScale: true,
+        width: 400,
+        height: 200,
+        padding: 0,
+        hideOnOverlayClick: false,
+        hideOnContentClick: false,
+        showCloseButton: false
+    });
+
+
+    $('#live_edit_feedback_str').html('');
+    // add liveToken in each link to navigate into the shop and keeping the liveedit mode
+    $('a').each(function () {
+        var href = this.href;
+        var search = this.search;
+        var hrefAdd = 'live_edit&liveToken=' + get('liveToken') + '&ad=' + get('ad') + '&id_shop=' + get('id_shop') + '&id_employee=' + get('id_employee');
+
+        if (href !== undefined && href !== '#' && href.substr(0, baseDir.length) == baseDir && !$(this).hasClass('settingModule')) {
+            if (search.length === 0)
+                this.search = hrefAdd;
+            else
+                this.search += '&' + hrefAdd;
+        }
+    });
+    // populate
+    $('.dndHook').each(function () {
+        hooks_list.push($(this).attr('id'));
+    });
+
+    $('.dndModule').each(function (index) {
+
+        var moduleBlock = $(this);
+
+        moduleBlock.attr('data-order', index);
+
+        var ids = moduleBlock.attr('id').split('_');
+
+        var name_module = ids['5'];
+
+        name_module = name_module.replace("-", "_");
+
+        modules_list.push(name_module);
+
+        var toolbar = $(this).find('.toolbar').html();
+
+        moduleBlock.find('.toolbar').remove();
+
+        moduleBlock.find(' > *').prepend('<div class="toolbar">' + toolbar + '</div>');
+
+        var ClassFirstElement = moduleBlock.find(' > *').attr('class');
+
+        if (ClassFirstElement !== undefined) {
+            var ClassArray = ClassFirstElement.split(" ");
+            $.each(ClassArray, function (key, value) {
+                var col = value.indexOf("col-");
+                if (col !== -1) {
+                    moduleBlock.addClass(value);
+                    moduleBlock.find(' > *').removeClass(value);
+                }
+            });
+        }
+    });
+
+    getHookableList();
+
+    $('.unregisterHook').unbind('click').click(function () {
+        var ids = $(this).attr('id').split('_');
+        $(this).closest(".toolbar").parent().parent().fadeOut('slow', function () {
```

Comment:
This is an unreliable and ugly way to select an element... No other solution ? (maybe not... please try)